Bug: recover_signer nightly fail

Will be fixed by https://github.com/ethereum/execution-specs/pull/1107

https://github.com/kkrt-labs/keth/actions/runs/13104028721/job/36555986795

```
FAILED tests/ethereum/cancun/test_transactions.py::TestTransactions::test_recover_sender - Failed: DID NOT RAISE <class 'Exception'>
Falsifying example: test_recover_sender(
    self=<test_transactions.TestTransactions object at 0x7f5a5ce2b130>,
    cairo_run_py=_run,
    chain_id=U64(2692),
    tx=FeeMarketTransaction(chain_id=U64(43),
     nonce=U256(1_782_048_801_376_639_692),
     max_priority_fee_per_gas=Uint(U64(27788)),
     max_fee_per_gas=Uint(U64(46)),
     gas=Uint(U64(32463)),
     to=b'\xb6\xa6\x97\x99<f\x15\x87-\xc3\x95\xc6\xe6\xde\x97\xbd\x00\x00\x00\x00',
     value=U256(52407),
     data=b'6/\xda\x80S\xb1\r4',
     access_list=((b'\x1b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
       (b'\x07\xfd\xc3\xd0\x9e\xbf\xce\xe2\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
        b'\x84\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')),
      (b'\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
       (b'\x07\xfd\xc3\xd0\x9e\xbf\xce\xe2\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
        b'\x84\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'))),
     y_parity=U256(2),
     r=U256(10231),
     s=U256(34211)),
)

You can reproduce this example by temporarily adding @reproduce_failure('6.124.3', b'AXicc+JqcWTKMNO/1RC8kdfEUdtDYrf0x6JyqTNOOT2Oek515x0ZAxn2Tr/37NjUw7rtomk2M6cv2+bMcGa7o7Qnw6Nz++ddOPyX3QlkCBrXSf27M0PrYgBjACjH') as a decorator on your test case
============ 1 failed, 851 passed, 2 skipped in 9203.47s (2:33:23) =============
```

--- a/cairo/ethereum/cancun/transactions.cairo
+++ b/cairo/ethereum/cancun/transactions.cairo
@@ -299,7 +299,7 @@ func recover_sender{
         let y_parity = tx.value.access_list_transaction.value.y_parity;
         let y_parity_is_zero = U256__eq__(y_parity, zero);
         let y_parity_is_one = U256__eq__(y_parity, U256(new U256Struct(low=1, high=0)));
-        with_attr error_message("ValueError") {
+        with_attr error_message("InvalidSignatureError") {
             assert (1 - y_parity_is_zero.value) * (1 - y_parity_is_one.value) = 0;
         }
         let hash = signing_hash_2930(tx.value.access_list_transaction);
@@ -312,7 +312,7 @@ func recover_sender{
         let y_parity = tx.value.fee_market_transaction.value.y_parity;
         let y_parity_is_zero = U256__eq__(y_parity, zero);
         let y_parity_is_one = U256__eq__(y_parity, U256(new U256Struct(low=1, high=0)));
-        with_attr error_message("ValueError") {
+        with_attr error_message("InvalidSignatureError") {
             assert (1 - y_parity_is_zero.value) * (1 - y_parity_is_one.value) = 0;
         }
 
@@ -326,7 +326,7 @@ func recover_sender{
         let y_parity = tx.value.blob_transaction.value.y_parity;
         let y_parity_is_zero = U256__eq__(y_parity, zero);
         let y_parity_is_one = U256__eq__(y_parity, U256(new U256Struct(low=1, high=0)));
-        with_attr error_message("ValueError") {
+        with_attr error_message("InvalidSignatureError") {
             assert (1 - y_parity_is_zero.value) * (1 - y_parity_is_one.value) = 0;
         }
         let hash = signing_hash_4844(tx.value.blob_transaction);

